### ISSUE TITLE
Service Button Widgets Do Not Open Authentication Window On Android 14 QPR1+

### ISSUE THREAD

**Issue by Nerwyn** [home-assistant/android#3893]
**Home Assistant Android app version(s):**
2023.9.2-full

**Android version(s):**
U1B1.230908.003 (Android 14 QPR 1)

**Device model(s):**
Pixel Fold

**Home Assistant version:**
2023.9.3

**Last working Home Assistant release (if known):**
2023.9.2 (HAOS)
2023.9.2-full (Android)

**Description of problem, include YAML if issue is related to notifications:**
Android service button widgets that have authentication enabled no longer work on Android 14 QPR Beta 1. Clicking the widget does nothing, but spam clicking the widget may inconsistently bring up the authentication window. Disabling authentication allows the widget to work (albeit without authentication). Service button widgets with authentication did work on Android 14 Beta 5.3. Recreating the widget or restarting the device does not fix the issue.

**Companion App Logs:**
The last bit of the attached companion app log seems relevant. Widget authentication is cancelling before authentication pops up.
```
09-26 21:10:26.203  4534  4534 D ButtonWidget: Broadcast received: 
09-26 21:10:26.203  4534  4534 D ButtonWidget: Broadcast action: io.homeassistant.companion.android.widgets.button.ButtonWidget.CALL_SERVICE_AUTH
09-26 21:10:26.203  4534  4534 D ButtonWidget: AppWidgetId: 163
09-26 21:10:26.203  4534  4534 D ButtonWidget: Calling authentication, then configured service
09-26 21:10:26.237  4534  4534 D WidgetAuthenticationA: onCreate in WidgetAuthenticationActivity called
09-26 21:10:26.260  4534  4534 D Unlock  : onAuthenticationError -> 10 :: Authentication canceled
09-26 21:10:26.260  4534  4534 D WidgetAuthenticationA: Authentication canceled by user, closing activity
09-26 21:10:26.269  4534  4534 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=androidx.activity.OnBackPressedDispatcher$Api33Impl$$ExternalSyntheticLambda0@5b589b6
09-26 21:10:26.294  4534  4534 W WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.app.Activity$$ExternalSyntheticLambda0@b86b8d5
09-26 21:10:27.644  4534  4534 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1695777025122, currentMillis: 1695777027644)
09-26 21:10:27.668  4534  4534 W Choreographer: Frame time is 0.153961 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.
09-26 21:10:27.698  4534  4534 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1695777025122, currentMillis: 1695777027698)
09-26 21:10:27.698  4534  4534 D IntegrationRepository: setAppActive(): true
09-26 21:10:29.180  4534 21357 D LogcatReader: Read logcat for pid 4534
```
[homeassistant_companion_log_8-26-2023_21-10-37.txt](https://github.com/home-assistant/android/files/12733066/homeassistant_companion_log_8-26-2023_21-10-37.txt)

**Screenshot or video of problem:**

https://github.com/home-assistant/android/assets/43071698/b7c2ab62-ac3e-408c-bef3-0b0426c60980

**Additional information:**
This may be an issue with Android 14 QPR Beta 1 and a bug for it has also been created in the Google issue tracker: https://issuetracker.google.com/issues/302159855

**Comment by jpelgrom** [comment]
Looks like the system doesn't like the theme, even when changing it to use `Theme.MaterialComponents.DayNight.NoActionBar` as a parent? Removing the theme completely does seem to 'fix' it but that breaks the illusion that only the authentication dialog is shown.

I keep seeing the following error after the activity's onCreate but before it returns the error code. Seems like a beta bug to me. I'd suggest waiting to see if this changes in the future before changing the appearance, removing one or two items from our theme didn't seem to help.

```
2023-09-27 22:04:05.796  3077-3077  AuthController          com.android.systemui                 D  showAuthenticationDialog, authenticators: 33023, sensorIds: 0 , credentialAllowed: true, requireConfirmation: false, operationId: 0, requestId: 8
2023-09-27 22:04:05.803  3077-3077  ThemeUtils              com.android.systemui                 E  View class com.android.systemui.biometrics.BiometricPromptLottieViewWrapper is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
```

**Comment by dshokouhi** [comment]
@Nerwyn now that QPR1 is out of beta is this still an issue for you? If so are you on the QPR2 beta or the production build?

**Comment by lawrencemaroun** [comment]
I just updated my pixel 7 pro to android 14 and I am experiencing this issue. Any known workaround before the bug is fixed?

**Comment by Zaca974** [comment]
i confirm the issue on pixel 7 with android 14

**Comment by jpelgrom** [comment]
Had another look at this now that the bug is also present in stable and looks like it has nothing to do with the theme as previously mentioned - it's only a log statement tagged as error but nothing's breaking because of it.

What I found above the previously found log statement is this line:

```
2023-12-19 22:41:25.754  2128-2128  AuthController          com.android.systemui                 W  Evicting client due to: com.google.android.apps.nexuslauncher
```

which leads to [this line](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/packages/SystemUI/src/com/android/systemui/biometrics/AuthController.java;l=240?q=authcontroller) -> function which is called [whenever the task stack changes](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/packages/SystemUI/src/com/android/systemui/biometrics/AuthController.java;l=191?q=authcontroller) -> a commit from June which [changes when the dialog is actually created and/or cancelled because of foreground state](https://cs.android.com/android/_/android/platform/frameworks/base/+/627e12ca86125eee82d8b7fd2a79a9a2314cab86) (as the source is now published). After testing with delays and listeners, if the app waits until the window is in the foreground (even though is transparent) instead of immediately authenticating, it will work again 🎉

TLDR: fix incoming.

**Comment by Nerwyn** [comment]
Got the latest Android app beta this morning. Can confirm that this issue is fixed.
